Tests/NIOTests/ChannelTests.swift

```diff
@@ -2214,6 +2214,30 @@ public class ChannelTests: XCTestCase {
         XCTAssertNoThrow(try sc.closeFuture.wait())
         XCTAssertNoThrow(try sc.syncCloseAcceptingAlreadyClosed())
     }
+
+    func testConnectWithECONNREFUSEDGetsTheRightError() throws {
+        let group = MultiThreadedEventLoopGroup(numThreads: 1)
+        defer {
+            XCTAssertNoThrow(try group.syncShutdownGracefully())
+        }
+        let serverSock = try Socket(protocolFamily: PF_INET, type: Posix.SOCK_STREAM)
+        // we deliberately don't set SO_REUSEADDR
+        XCTAssertNoThrow(try serverSock.bind(to: SocketAddress(ipAddress: "127.0.0.1", port: 0)))
+        let serverSockAddress = try! serverSock.localAddress()
+        XCTAssertNoThrow(try serverSock.close())
+
+        for _ in 0..<64 {
```

Comment:
Do we need the loop? We fail in all cases other than the throw, which suggests we always throw.

Reply:
@Lukasa in my tests it made the probability that we also hit the async connect path a lot higher (as in pretty much always). Therefore I thought that's a good idea :)